### PR TITLE
Use containers/storage/pkg/regexp to better handle init regex

### PR DESCRIPTION
**Pull request by rhatdan**
Use github.com/containers/storage/pkg/regexp to better handle init regex
    
    The new Reexp structure, will handle the initialization of regexp
    and then will do the MustCompile within a sync.Once field.
    
Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

### CODE REVIEW

**Comment by nalind** [comment]
LGTM, but unless there's a need to do otherwise, I prefer vendoring tagged versions of dependencies.  If we can wait for a tagged release of containers/storage that includes this API, I'd like to do that.

**Comment by rhatdan** [comment]
I cut a release to move this along.

@flouthoc PTAL

**Review by flouthoc — APPROVED** [review]
LGTM

**Comment by nalind** [comment]
Looks like #234 started triggering errors in Travis for a conformance test that was added in #222 but which was failing.  We may need to revisit #219.

**Comment by rhatdan** [comment]
@giuseppe PTAL, looks like your change might be breaking tests?

**Comment by rhatdan** [comment]
Looks like https://app.travis-ci.com/github/openshift/imagebuilder/jobs/593695010 is having the same problem.

**Comment by nalind** [comment]
If I'm reading https://app.travis-ci.com/github/openshift/imagebuilder/jobs/595513477#L668 right, `docker build` produced an image which included `/var/www/file` and `/var/www/file2`, but the image that `imagebuilder` built using the same sources didn't include those.  But when I run the test with Go 1.20 and its version of moby-engine on my Fedora 37 box, the test passes.  Restarted the job to see if it's not just a weird flake.

**Pull request merged**
